dissect/hypervisor/backup/vma.py

```diff
@@ -164,7 +163,7 @@ def __init__(self, fh, offset):
             elif cluster_num > self._max[dev_id]:
                 self._max[dev_id] = cluster_num
 
-            self.blocks[dev_id].append((dev_id, cluster_num, mask))
+            self.blocks.append((dev_id, cluster_num, mask))
```

Comment:
I don't yet see how this change affects anything, can you walk me through it or provide an example with e.g. some mock tuples?

In the snippet later on, you iterate over all entries in `extent.blocks`, does that not achieve the same as what the original code did, with the "filtering" being done in the dictionary? You write "inside an extent, blocks of multiple devices could be stored", but doesn't it already support that by storing those blocks in separate dictionary keys?

Reply:
Sorry for any confusion that a lack of explanation from our side may have caused. Image the following extent: 
```
|--------|------------|--------|
| dev_id | cluster_no | mask   |
|--------|------------|--------|
|      1 |          0 | 0xFFFF |
|      1 |          1 | 0xFFFF |
|      2 |          0 | 0xFFFF |
|      1 |          2 | 0xFFFF |
|    ... |        ... |    ... |
|--------|------------|--------|
```
The function `_iter_clusters(vma, device=2, cluster=0, count=1)` should yield `(0, 0xFFFF, 2 * 16 * c_vma.VMA_BLOCK_SIZE)`. The old code returned `(0, 0xFFFF, 0)` instead, since the `block_offset` was not incremented for the first two clusters in this extent (because the loop only iterated over `extent.blocks[dev_id]`, thus missing the first blocks).

Reply:
Aaaah, the devil is in the `block_offset` calculation, I see it now, thanks!

I then wonder if this code would be simpler if we moved the `block_offset` calculation to the loop in the `Extent` class. What's your opinion on that?

Reply:
Sounds reasonable! This would also make the results of the offset calculations being stored in the LRU cache, isn't it?

Reply:
Yes, I believe so!